src/client/app/components/ChartSelectComponent.tsx

```diff
@@ -3,85 +3,62 @@
 /**
  *  A component that allows users to select which chart should be displayed.
  * @returns Chart select element
  */
 export default function ChartSelectComponent() {
-	const divBottomPadding: React.CSSProperties = {
-		paddingBottom: '15px'
-	};
-	const labelStyle: React.CSSProperties = {
-		fontWeight: 'bold',
-		margin: 0
-	};
-
-	const dispatch: Dispatch = useDispatch();
+	const currentChartToRender = useAppSelector(selectChartToRender)
+	const dispatch = useAppDispatch();
 	const [expand, setExpand] = useState(false);
-	const sortedMaps = _.sortBy(_.values(useSelector((state: State) => state.maps.byMapID)).map(map => (
-		{ value: map.id, label: map.name, isDisabled: !(map.origin && map.opposite) } as SelectOption
-	)), 'label');
+
+	// TODO Re-write as selector to use elsewhere
+	// const sortedMaps = _.sortBy(_.values(useSelector((state: State) => state.maps.byMapID)).map(map => (
+	// 	{ value: map.id, label: map.name, isDisabled: !(map.origin && map.opposite) } as SelectOption
+	// )), 'label');
 
 	return (
-		<div style={divBottomPadding}>
+		<>
 			<p style={labelStyle}>
 				<FormattedMessage id='graph.type' />:
 				<TooltipMarkerComponent page='home' helpTextId='help.home.chart.select' />
 			</p>
 			<Dropdown isOpen={expand} toggle={() => setExpand(!expand)}>
 				<DropdownToggle outline caret>
-					<FormattedMessage id={useSelector((state: State) => state.graph.chartToRender)} />
+					<FormattedMessage id={currentChartToRender} />
 				</DropdownToggle>
 				<DropdownMenu>
-					<DropdownItem
-						onClick={() => dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.bar })}
-					>
-						<FormattedMessage id='bar' />
-					</DropdownItem>
-					<DropdownItem
-						onClick={() => dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.compare })}
-					>
-						<FormattedMessage id='compare' />
-					</DropdownItem>
-					<DropdownItem
-						onClick={() => dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.line })}
-					>
-						<FormattedMessage id='line' />
-					</DropdownItem>
-					<DropdownItem
-						onClick={() => {
-							dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.map });
-							if (Object.keys(sortedMaps).length === 1) {
-								// If there is only one map, selectedMap is the id of the only map. ie; display map automatically if only 1 map
-								dispatch({ type: 'UPDATE_SELECTED_MAPS', mapID: sortedMaps[0].value });
-							}
-						}}
-					>
-						<FormattedMessage id='map' />
-					</DropdownItem>
-					<DropdownItem
-						onClick={() => dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.radar })}
-					>
-						<FormattedMessage id='radar' />
-					</DropdownItem>
-					<DropdownItem
-						onClick={() => dispatch({ type: 'CHANGE_CHART_TO_RENDER', chartType: ChartTypes.threeD })}
-					>
-						<FormattedMessage id='3D' />
-					</DropdownItem>
+					{
+						// Make items for dropdown from enum
+						Object.values(ChartTypes)
```

Comment:
I like the new code but the items should be alphabetical.

Reply:
I added a .sort() to the map method. Let me know if this works for you. Otherwise we can rollback.

Reply:
looks good